compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -4352,7 +4537,7 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
           cpy.Ident(qual)(qual.symbol.name.sourceModuleName.toTypeName)
         case _ =>
           errorTree(tree, em"cannot convert from $tree to an instance creation expression")
-      val tycon = ctorResultType.underlyingClassRef(refinementOK = false)
+      val tycon = ctorResultType.underlyingClassRef(refinementOK = true)
```

Comment:
No need to check if the x.modularity is enabled here?

Reply:
yes, good catch.